### PR TITLE
[ETFE-3498] - Added handling for degrees plato in error (704)

### PR DESCRIPTION
**Pull request by LewisMitchell**
~Raising as draft as it needs refactoring~

Added in some missing tests on SAD and handling where a user navigates to declaration manually but they haven't fixed all submission failures

### CODE REVIEW

**Comment by LewisMitchell** [comment]
Scenarios fixed in this PR:
- when a user has 1 error for item 1, and 2 errors for item 2, before if they removed item 1, only 1 of the 2 errors for item 2 would be reindexed
- when 2 items are in error, and the first item is changed (EPC code) it  would reindex all other items, however it should just remove the errors for that item and not touch any others

File: app/utils/SubmissionFailureErrorCodes.scala [hunk 1]
```diff
@@ -16,9 +16,52 @@
...
+  sealed trait ErrorCode {
+    val code: String
+    val messageKey: String
+    val id: String
+    def route()(implicit request: DataRequest[_]): Call
+    val index: Option[Index] = None
+  }
+
+  case class ItemQuantityError(idx: Index, isForAddToList: Boolean) extends ErrorCode {
+    override val code = "4402"
```
Comment (LewisMitchell) [inline app/utils/SubmissionFailureErrorCodes.scala:41]:
```suggestion
    override val code = itemQuantityError
```

File: app/utils/SubmissionFailureErrorCodes.scala [hunk 2]
```diff
@@ -16,9 +16,52 @@
...
+    override val messageKey = s"errors.704.items.quantity${if(isForAddToList) ".addToList" else ""}"
+    override val id = s"fix-item-${idx.displayIndex}-quantity"
+    override def route()(implicit request: DataRequest[_]): Call =
+      controllers.sections.items.routes.ItemQuantityController.onPageLoad(request.ern, request.draftId, idx, CheckMode)
+    override val index = Some(idx)
+
+  }
+
+  case class ItemDegreesPlatoError(idx: Index, isForAddToList: Boolean) extends ErrorCode {
+    override val code = "4451"
```
Comment (LewisMitchell) [inline app/utils/SubmissionFailureErrorCodes.scala:51]:
```suggestion
    override val code = itemDegreesPlatoError
```

**Review by james-work-account — CHANGES_REQUESTED** [review]
Some of the files are a little strangely formatted, could you run an auto-format on the (non-view) files you've touched please?

Otherwise LGTM

**Review by james-work-account — APPROVED** [review]

**Pull request merged**